include/boost/spirit/home/karma/detail/output_iterator.hpp

```diff
@@ -383,7 +383,7 @@ namespace boost { namespace spirit { namespace karma { namespace detail
           , output_iterator<OutputIterator, Properties, Derived>
         >::type most_derived_type;
 
-        enum { properties = Properties::value };
+        static const int properties = Properties::value;
```

Comment:
Try making it `generator_properties::enum_type` instead of `int`.

Reply:
I just did. Waiting for the CI to confirm it didn't break anything.

Reply:
Apparently it doesn't work without a static cast, so I will add it.

I am actually unsure you actually want the `enum_type` or the `std:: underlying_type_t<enum_type>` given that in `Properties::value` you won't have actual enum values, but a conjunction of enum value bitwise-and'ed together. As you wish.